hooks/src/useFormattedCardNumber/useFormattedCardNumber.ts

```diff
@@ -0,0 +1,55 @@
+import {
+  CARD_NETWORK_PATTERNS,
+  TypeOfCardNetworkPatterns,
+} from "../constants/cardNetworkPatterns";
+
+type CardNumbers = {
+  FIRST: string;
+  SECOND: string;
+  THIRD: string;
+  FOURTH: string;
+};
+
+export default function useFormattedCardNumbers(
+  cardNumbers: CardNumbers,
+  userCardNetworkFormatter?: TypeOfCardNetworkPatterns
+) {
+  const fullCardNumber =
+    `${cardNumbers.FIRST}${cardNumbers.SECOND}${cardNumbers.THIRD}${cardNumbers.FOURTH}`.replace(
+      /\D/g,
+      ""
+    );
+  const patterns = { ...CARD_NETWORK_PATTERNS, ...userCardNetworkFormatter };
+
+  const matchedNetwork = Object.entries(patterns).find(([_, { PATTERN }]) =>
+    PATTERN.test(fullCardNumber)
+  );
+
+  if (!matchedNetwork)
+    return {
+      formattedCardNumbers: "",
+      error: true,
+      errorMessage: "카드 번호를 포맷팅할 수 없습니다.",
+    };
+
+  const [parsedCardNetwork, { FORMAT_PATTERN, LENGTH, FORMAT_TEMPLATE }] =
+    matchedNetwork;
```

Comment:
parsedCardNetwork 는 사용되지 않는 변수로 보이는데요, 이러한 경우엔 다음과 같이 작성하여 변수 선언을 생략할 수 있습니다.

```suggestion
  const [, { FORMAT_PATTERN, LENGTH, FORMAT_TEMPLATE }] =
    matchedNetwork;
```

Reply:
다음과 같이 작성하는 것 또한 유효합니다.

```suggestion
  const [_, { FORMAT_PATTERN, LENGTH, FORMAT_TEMPLATE }] =
    matchedNetwork;
```

Reply:
[f8d7080](https://github.com/woowacourse/react-modules/pull/130/commits/f8d70802a4de7b6bb1d52fd62b6bd2fa1b82173b)

앗 구현하면서 놓친 부분이네용 수정하였습니다!